src/app/content/components/Sidebar.tsx

```diff
@@ -219,43 +362,84 @@ export class Sidebar extends Component<SidebarProps> {
 
   public componentDidUpdate() {
     this.scrollToSelectedPage();
+    this.expandCurrentChapter();
   }
 
   private scrollToSelectedPage() {
-    scrollTocSectionIntoView(this.sidebar, this.activeSection);
+    scrollTocSectionIntoView(this.sidebar.current, this.activeSection.current);
   }
 
-  private renderTocNode = (book: Book, {contents}: ArchiveTree) => <nav>
-    <ol>
-      {contents.map((item) => {
-        const active = (!!this.props.page) && stripIdVersion(item.id) === this.props.page.id;
+  private expandCurrentChapter() {
 
-        return isArchiveTree(item)
-          ? <NavItem key={item.id}>
-              <h3 dangerouslySetInnerHTML={{__html: item.title}} />
-              {this.renderTocNode(book, item)}
-            </NavItem>
-          : <NavItem
-              key={item.id}
-              ref={active ? ((ref: any) => this.activeSection = ref) : undefined}
-              active={active}
-            >
-              <ContentLink
-                onClick={this.props.onNavigate}
-                book={book}
-                page={item}
-                dangerouslySetInnerHTML={{__html: item.title}}
-              />
-            </NavItem>;
-      })}
-    </ol>
-  </nav>
+    if (typeof(document) === 'undefined' || typeof(window) === 'undefined') {
+      return;
+    }
 
-  private renderTocHeader = () => <ToCHeader>
-    <SidebarHeaderButton><TimesIcon /></SidebarHeaderButton>
-  </ToCHeader>
+    const currentChapter = this.activeSection.current;
+    let parent: HTMLElement;
+    let counter = 1;
 
-  private renderToc = (book: Book) => this.renderTocNode(book, book.tree);
+    if ( currentChapter && currentChapter.parentElement) {
+      parent = currentChapter.parentElement;
+
+      while ( parent.getAttribute('aria-label') !== 'Table of Contents'
+        && counter < 10) {
+        console.log(parent);
+
+        if ( parent.tagName === 'DETAILS'
+          && !parent.hasAttribute('open')
+        ) {
+          parent.setAttribute('open', '');
+        }
+
+        if ( parent.parentElement ) {
+          parent = parent.parentElement;
+        } else {
+          return null;
+        }
+        counter++;
+
+      }
+    }
+  }
+
+  private renderChildren = (book: Book, section: ArchiveTree) =>
+    <NavOl section={section}>
+      {section.contents.map((item) => {
+        const active = (!!this.props.page) && stripIdVersion(item.id) === this.props.page.id;
+        return isArchiveTree(item)
+        ? <NavItem>{this.renderTocNode(book, item)}</NavItem>
+        : <NavItem
+          key={item.id}
+          ref={active ? this.activeSection : null}
+          active={active}
+        >
+          <ContentLink
+            onClick={this.props.onNavigate}
+            book={book}
+            page={item}
+            dangerouslySetInnerHTML={{__html: item.title}}
+          />
+        </NavItem>;
+      })}
+    </NavOl>
```

Comment:
semicolon

Reply:
lint on my editor says they're unnecessary, should I still add them?

Reply:
pushed a fix to the linter config